### PR TITLE
fix(curriculum): trim string to be working selector

### PR DESCRIPTION
**Pull request by majestic-owl448**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

Quick fix for the test on step 46 of the quiz project, if the string selector has leading or trailing spaces, it doesn't work as a selector

### CODE REVIEW

**Review by moT01 — COMMENTED** [review]
This works for space around the `a`, but it doesn't work if there's spaces around any of the other parts of the selector. For example:

```
li  > a {
  color: #dfdfe2;
}
```

^ two spaces after the `li`. This PR is an improvement, so I would be fine with it going in - but do we want to address that at all @ilenia-magoni?

File: curriculum/challenges/english/14-responsive-web-design-22/learn-accessibility-by-building-a-quiz/614796cb8086be482d60e0ac.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ You should give the `a` elements inside list items a `color` property.
 const helper = new __helpers.CSSHelp(document);
 
 const usedSelector = code.match(/^\s*?(.*?a)\s*?{/m)[1];
-assert.notEmpty(helper.getStyle(usedSelector)?.color);
+assert.notEmpty(helper.getStyle(usedSelector.trim())?.color);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-accessibility-by-building-a-quiz/614796cb8086be482d60e0ac.md:29]:
```suggestion
assert.notEmpty(helper.getStyle(usedSelector.replaceAll(/\s+/, ' ').trim())?.color);
```

**Comment by ilenia-magoni** [comment]
What about now @moT01 ?

File: curriculum/challenges/english/14-responsive-web-design-22/learn-accessibility-by-building-a-quiz/614796cb8086be482d60e0ac.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ You should give the `a` elements inside list items a `color` property.
 const helper = new __helpers.CSSHelp(document);
 
 const usedSelector = code.match(/^\s*?(.*?a)\s*?{/m)[1];
-assert.notEmpty(helper.getStyle(usedSelector)?.color);
+assert.notEmpty(helper.getStyle(usedSelector.replaceAll(/\s+/, ' ').trim())?.color);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-accessibility-by-building-a-quiz/614796cb8086be482d60e0ac.md:29]:
```suggestion
assert.notEmpty(helper.getStyle(usedSelector.replaceAll(/\s+/g, ' ').trim())?.color);
```

**Review by Sembauke — APPROVED** [review]

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**